app/containers/UIKit/DatePicker.tsx

```diff
@@ -48,11 +48,15 @@ export const DatePicker = ({ element, language, action, context, loading, value,
 	// timestamp as number exists in Event
 	// @ts-ignore
 	const onChange = ({ nativeEvent: { timestamp } }: Event, date?: Date) => {
-		const newDate = date || new Date(timestamp);
-		onChangeDate(newDate);
-		action({ value: moment(newDate).format('YYYY-MM-DD') });
-		if (isAndroid) {
-			onShow(false);
+		if (date || timestamp) {
+			const newDate = date || new Date(timestamp);
+			unstable_batchedUpdates(() => {
```

Comment:
We need this `unstable_batchedUpdates` to avoid double re-render of the calendar when using Android